.github/workflows/check-unity-sdk-for-updates.yml

```diff
@@ -0,0 +1,55 @@
+name: Check Unity SDK Updates
+
+on:
+  schedule: # Run Hourly
+    - cron: '0 * * * *'
+  workflow_dispatch:
+
+jobs:
+  check-updates:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.PUBLIC_REPO_BOT_GITHUB_TOKEN }}
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Check for AdGem SDK updates
+        run: |
+          npm run check-adgem-version
+          echo "HAS_CHANGES=$(git status --porcelain -z)" >> $GITHUB_ENV
+        env:
+          GITHUB_TOKEN: ${{ secrets.PUBLIC_REPO_BOT_GITHUB_TOKEN }}
+
+      - name: Commit and push changes
+        if: env.HAS_CHANGES
+        run: |
+          git config --global user.name "${{ vars.BOT_NAME || 'adgem-developers' }}"
+          git config --global user.email "${{ vars.BOT_EMAIL || 'developers@adgem.com' }}"
+          git add .
+          git commit -m 'build: update adgem sdk dependency version'
+          git push --force origin HEAD:refs/heads/update-adgem-sdk
```

Comment:
i think the --force could cause problems

Reply:
I can remove it, but I think it might be necessary to actually push a new branch like this.